### ISSUE TITLE
Global names mapping via convictions

### ISSUE THREAD

**Issue by DerHulk** [TinyMapper/TinyMapper#41]
Hello, I have to say that the TinyMapper really is a good alternative to the AutoMapper and your good work done.
Unfortunately, I miss a feature where you can put global names mapping via convictions. e.g.
```
ObjectA.Name -> ObjectB.Name1,
ObjectA.Titel -> ObjectB.Titel1
```
I have writed some code ( [my fork with changes](https://github.com/DerHulk/TinyMapper/commit/cb1666fb52b8b320387137908bc45ce7739211c1) , [bug fix :-) ](https://github.com/DerHulk/TinyMapper/commit/3feeed5448bc0b3eceeaf6200b71332617a6f232)) and maybe 
you will take it into the TinyMapper or you implementate a better solution for this kind of mapping in the Tinymapper.

Thanks.

**Comment by GSerjo** [comment]
DerHulk, look really interesting :)

You can do similar thru `config.Bind`, but if you have a lot of similar fields (`Name -> Name1 etc`)  your approach is much better

```
TinyMapper.Bind<Person, PersonDto>(config =>
{
    config.Bind(source => source.LastName, target => target.Surname);
});
```
Let's add this feature :)

**Comment by DerHulk** [comment]
Hi GSerjo, 
best message of the day great to here that  👍  . 
Something what i can do?

**Comment by GSerjo** [comment]
glad to hear :) sure, you can create a pull request and it will be a starting point

**Comment by DerHulk** [comment]
Ok, done ;-)

**Comment by GSerjo** [comment]
DerHulk,  updated a bit your version. what do you think about this one?

```
        public void CustomeNameMatching_Success()
        {
            TinyMapper.Config(config =>
            {
                config.NameMatching((x, y) => string.Equals(x + "1", y, StringComparison.OrdinalIgnoreCase));
            });

            var sourceCustom = new SourceCustom { Custom = "Hello", Default = "Default" };
            var targetCustom = TinyMapper.Map<TargetCustom>(sourceCustom);

            Assert.Equal(sourceCustom.Custom, targetCustom.Custom1);
            Assert.True(string.IsNullOrEmpty(targetCustom.Default));

            TinyMapper.Config(config =>
            {
                config.Reset();
            });

            var sourceDefault = new SourceDefault { Custom = "Hello", Default = "Default" };
            var targetDefault = TinyMapper.Map<TargetDefault>(sourceDefault);

            Assert.True(string.IsNullOrEmpty(targetDefault.Custom1));
            Assert.Equal(sourceDefault.Default, targetDefault.Default);
        }
```

**Comment by DerHulk** [comment]
Great looks better, so we you have checked the Reset too 👍  . I would recommand to run this test in a sepparate appdomain so no other test is affected by the global change (for example in parallel running). One way to archiv this could be [here](http://stackoverflow.com/questions/21237886/how-can-i-use-a-separate-appdomain-for-each-xunit-net-test-method).

**Comment by GSerjo** [comment]
sure, agree with you :)

**Comment by DerHulk** [comment]
Ok, i will try to create a new pull request for this in the next few days :-).
